test/e2e/framework/util.go

```diff
@@ -1342,6 +1344,7 @@ func ExpectNoErrorWithRetries(fn func() error, maxRetries int, explain ...interf
 		}
 		e2elog.Logf("(Attempt %d of %d) Unexpected error occurred: %v", i+1, maxRetries, err)
 	}
+	debug.PrintStack()
```

Comment:
should this PrintStack call should only happen if err != nil ?

Reply:
@neolit123  For wrappers like `ExpectEqual` and `ExpectNotEqual`, we need to put the `PrintStack` into a condition statement to avoid it dump the stacktrace every time the wrappers be called. But `gomega.Expect()` has no return value. I don't have any better insights to do some change in the wrappers. But I do something similar like this: For those tests that call `ExpectEqual/ExpectNotEqual`, have a `try..catch` thing. If exception happens, then call `PrintStack()`. But Go doesn't have this kind of mechanism so this may involve a lot of changes. Or do you have any insights ?

Reply:
Capture the stdout/stderr after `ExpectEqual/ExpectNotEqual` is also a potential solution. Need to implement a wrappers for the capture behavior.

Reply:
sorry my comment here was about this.
> should this PrintStack call should only happen if err != nil ?

```
if err != nil {
  debug.PrintStack()
}
```

otherwise we are printing a stack trace even if there is no error in the retires.

Reply:
ready to re-review